module/VuFind/src/VuFind/OAI/Server.php

```diff
@@ -444,24 +445,18 @@ protected function attachNonDeleted(
         $headerOnly = false,
         $set = ''
     ) {
-        // Get the XML (and display an error if it is unsupported):
         if ($format === false) {
-            $xml = '';      // no metadata if in header-only mode!
-        } elseif ('oai_vufind_json' === $format && $this->supportsVuFindMetadata()) {
-            $xml = $this->getVuFindMetadata($record);   // special case
-        } else {
-            $xml = $record
-                ->getXML($format, $this->baseHostURL, $this->recordLinkerHelper);
-            if ($xml === false) {
-                return false;
-            }
+            return true;
```

Comment:
```suggestion
            // If no format was requested, report success without doing anything:
            return true;
```

Reply:
Thanks! I took the comment and added it in without commit. :) I also adjusted the comment, which was previously set, as it was littlebit unclear for my own taste.